controller/state/statedb.go

```diff
@@ -752,7 +752,7 @@ func (s *stateDB) PublishRecordsFrom(ctx context.Context, worker string) error {
 			if err != nil {
 				return err
 			}
-			tskBuilder := tasks.Type.FinishedTask.NewBuilder()
+			tskBuilder := tasks.Type.FinishedTask__Repr.NewBuilder()
```

Comment:
This seems quite unrelated? What does it fix?

Reply:
I think it was either @mvdan or @rvagg  who mentioned that we should be de-serializing into __Repr's as well as serializing into them. This is a remnant of the constant 'the default schema'd types are not the ones you actually should be using problem in the current ipld-prime' 
in practice - it shouldn't matter any any valid representation _should_ de-serialize into the nonrepresentation type, but i think that might not be guaranteed?

Reply:
This was me, yes. In practice, in this particular case, it might not matter. But it's certainly asking for trouble. This would definitely break if we added a struct field with non-default representation, for example.